detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivateMember.kt

```diff
@@ -118,19 +130,12 @@ private class UnusedFunctionVisitor(allowedNames: Regex) : UnusedMemberVisitor(a
      */
     override fun visitReferenceExpression(expression: KtReferenceExpression) {
         super.visitReferenceExpression(expression)
-        when (expression) {
-            is KtOperationReferenceExpression -> callExpressions.add(expression.getReferencedName())
-            is KtNameReferenceExpression -> callExpressions.add(expression.getReferencedName())
-        }
-    }
-
-    override fun visitCallExpression(expression: KtCallExpression) {
-        super.visitCallExpression(expression)
-        val calledMethodName = expression.referenceExpression()?.text
-
-        if (calledMethodName != null) {
-            callExpressions.add(calledMethodName)
-        }
+        val name = when (expression) {
+            is KtOperationReferenceExpression -> expression.getReferencedName()
+            is KtNameReferenceExpression -> expression.getReferencedName()
+            else -> null
+        } ?: return
+        functionReferences.getOrPut(name) { mutableListOf() }.add(expression)
```

Comment:
Could these statements be simplified to have a lower complexity? I'm thinking about something like this.

```suggestion
        if (expression is KtOperationReferenceExpression || expression is KtNameReferenceExpression) {
            val name = expression.getReferencedName()
            functionReferences.getOrPut(name) { mutableListOf() }.add(expression)
        }
```

Reply:
No. To call 'getReferencedName()', smart cast is required.

Reply:
Thanks for the explanation! :-)
resolved...